src/main/kotlin/org/rust/ide/utils/ExprUtils.kt

```diff
@@ -103,37 +103,46 @@ val RsUnaryExpr.operatorType: UnaryOperator?
 /**
  * Simplifies a boolean expression if can.
  *
+ * @param peek if true then does not perform any changes on PSI, 
+ *             `expr` is not defined and `result` indicates if this expression
+ *             can be simplified
  * @return `(expr, result)` where `expr` is a resulting expression,
  *         `result` is true if an expression was actually simplified.
  */
-fun RsExpr.simplifyBooleanExpression(): Pair<RsExpr, Boolean> {
+fun RsExpr.simplifyBooleanExpression(peek: Boolean = false): Pair<RsExpr, Boolean> {
```

Comment:
Fixed.
`simplifyBooleanExpression` still returns a pair to prevent twice traversal of PSI tree.